### PR TITLE
Added HB-01 implementation

### PR DESCRIPTION
**Pull request by oroboro-cpu**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
-target/*
\ No newline at end of file
+target/*
+hibernate.iml
```
Comment (yurwing) [inline .gitignore:4]: you may not write this, because you have `*.iml`

**Review by yurwing — APPROVED** [review]

File: src/main/resources/hibernate.cfg.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE hibernate-configuration PUBLIC
+        "-//Hibernate/Hibernate Configuration DTD 3.0//EN"
+        "http://hibernate.org/dtd/hibernate-configuration-3.0.dtd">
+<hibernate-configuration>
+    <session-factory>
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL5Dialect</property>
+        <property name="connection.url">jdbc:mysql://localhost:3306/cinema</property>
```
Comment (yurwing) [inline src/main/resources/hibernate.cfg.xml:8]: you didn't write a serverTimezone, but I don't now is it necessary here, but in the video it was

File: src/main/java/cinema/project/dao/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/cinema/project/dao/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/cinema/project/dao/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/cinema/project/dao/MovieDaoImpl.java]:
```suggestion
            throw new RuntimeException("Can't insert movie entity " + movie, e);
```

File: src/main/java/cinema/project/dao/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        } finally {
+            session.close();
```
Comment (Sofasmile) [inline src/main/java/cinema/project/dao/MovieDaoImpl.java]: let's add a check for `null` value

File: src/main/java/cinema/project/dao/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        }
+    }
+
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all movies from DB");
```
Comment (Sofasmile) [inline src/main/java/cinema/project/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all movies from DB", e);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (oroboro-cpu) [reply .gitignore:4]: I did it coz this file was added to the git repository even though *.iml

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**